runtime/src/message_processor.rs

```diff
@@ -302,22 +303,28 @@ impl<'a> ThisInvokeContext<'a> {
     }
 }
 impl<'a> InvokeContext for ThisInvokeContext<'a> {
-    fn push(&mut self, key: &Pubkey) -> Result<(), InstructionError> {
-        if self.program_ids.len() > self.bpf_compute_budget.max_invoke_depth {
+    fn push(
+        &mut self,
+        keyed_accounts: &[KeyedAccount],
+        key: &Pubkey,
+    ) -> Result<&[KeyedAccount], InstructionError> {
+        if self.invoke_stack.len() > self.bpf_compute_budget.max_invoke_depth {
             return Err(InstructionError::CallDepth);
         }
-        if self.program_ids.contains(key) && self.program_ids.last() != Some(key) {
+        let frame_index = self.invoke_stack.iter().position(|frame| frame.0 == *key);
```

Comment:
Was there a corner case here that was missing before?

Reply:
No, but it is a tuple / struct now so `contains` and `Some(key) `do not work anymore.